src/pgduckdb_detoast.cpp

```diff
@@ -106,15 +111,31 @@ ToastFetchDatum(struct varlena *attr) {
 		SET_VARSIZE(result, attrsize + VARHDRSZ);
 	}
 
-	if (attrsize == 0)
+	if (attrsize == 0) {
 		return result;
+	}
 
 	DuckdbProcessLock::GetLock().lock();
-	toastrel = table_open(toast_pointer.va_toastrelid, AccessShareLock);
-	table_relation_fetch_toast_slice(toastrel, toast_pointer.va_valueid, attrsize, 0, attrsize, result);
-	table_close(toastrel, AccessShareLock);
+	toast_rel = try_table_open(toast_pointer.va_toastrelid, AccessShareLock);
+
+	if (toast_rel != NULL) {
+		DuckdbProcessLock::GetLock().unlock();
+		throw duckdb::InternalException("(PGDuckDB/ToastFetchDatum) Error opening toast relation");
+	}
+
+	bool error_fetch_toast = false;
+	if (PostgresFunctionGuard(table_relation_fetch_toast_slice, toast_rel, toast_pointer.va_valueid, attrsize, 0,
+	                          attrsize, result).value()) {
+		error_fetch_toast = true;
+	}
+
+	table_close(toast_rel, AccessShareLock);
```

Comment:
Even `table_close` can throw some ERROR, see this quote from the PG source code where they consider memory context reset callbacks possibly throwing an error.

```
	/*
	 * It's not entirely clear whether 'tis better to do this before or after
	 * delinking the context; but an error in a callback will likely result in
	 * leaking the whole context (if it's not a root context) if we do it
	 * after, so let's do it before.
	 */
	MemoryContextCallResetCallbacks(context);
```

Reply:
Guarded.